src/v/cloud_storage_clients/s3_client.cc

```diff
@@ -428,9 +430,34 @@ rest_error_response parse_xml_rest_error_response(iobuf&& buf) {
     }
 }
 
+ss::future<rest_error_response> parse_unknown_format_error_response(
+  boost::beast::http::status status, iobuf buf) {
+    static constexpr auto max_body_size = static_cast<size_t>(4096U);
+    auto read_size = buf.size_bytes();
+
+    // log out if iobuf_to_string had to truncate the error message
+    if (read_size > max_body_size) {
+        vlog(
+          s3_log.info,
+          "error response body too long. found size {} truncating to size {}",
+          read_size,
+          max_body_size);
+        read_size = max_body_size;
+    }
+
+    iobuf_parser parser{std::move(buf)};
+    const auto error_body = parser.read_string(read_size);
+
+    co_return rest_error_response{
+      fmt::format("{}", status_to_error_code(status)),
+      fmt::format("http status: {}, error body: {}", status, error_body),
```

Comment:
you might consider adding the truncation notice here if the log was truncated, instead of using a dedicate INFO level log line for it. something like this

```
error body [truncated~4096]: ...
```